Frontend/src/Base/BaseImageUpload/BaseImageUpload.jsx

```diff
@@ -0,0 +1,102 @@
+import { useRef, useState } from 'react';
+import BaseButton from '../BaseButton';
+import { useTranslation } from 'react-i18next';
+import UploadedImagesDisplay from './UploadedImagesDisplay';
+import { useDispatch, useSelector } from 'react-redux';
+import { selectAddPlace, addPlaceActions } from '../../Redux/addPlaceSlice';
+
+const BaseImageUpload = ({ fileSize }) => {
+  const dispatch = useDispatch();
+  const images = useSelector(selectAddPlace).images;
+  const [errorMsg, setErrorMsg] = useState(null);
+  const [showError, setShowError] = useState(null);
+  const fileInputRef = useRef(null);
+  const { t } = useTranslation();
+  const MAX_FILE_SIZE = fileSize * 1024 * 1024;
+  const MAX_FILES = 3;
+
+  const handleClick = () => {
+    fileInputRef.current.click();
+  };
+
+  const validateImagesQuantity = (uploadedImagesLength) => {
+    const allFilesLenght = uploadedImagesLength + images.length;
+
+    if (uploadedImagesLength > MAX_FILES || allFilesLenght > MAX_FILES) {
+      setErrorMsg(`${t('common.max_files_number')} ${MAX_FILES}`);
+      return false;
+    } else {
+      return true;
+    }
+  };
+
+  const validateDuplicates = (uploadedImage) => {
+    const duplicateFile = images.find((image) => image.name === uploadedImage.name);
+    if (duplicateFile) {
+      setErrorMsg(`${t('common.already_uploaded')} ${uploadedImage.name}`);
+      return false;
+    } else {
+      return true;
+    }
+  };
+
+  const validateImageType = (uploadedImage) => {
+    if (!uploadedImage.type.startsWith('image/')) {
+      setErrorMsg(t('common.only_images'));
+      return false;
+    } else {
+      return true;
+    }
+  };
+
+  const validateImageSize = (uploadedImage) => {
+    if (uploadedImage.size >= MAX_FILE_SIZE) {
+      setErrorMsg(`${t('common.file_too_large')} ${uploadedImage.name}`);
+      return false;
+    } else {
+      return true;
+    }
+  };
+
+  const handleImageUpload = (event) => {
+    const uploadedImages = Array.from(event.target.files);
+
+    const isImagesQuantityValid = validateImagesQuantity(uploadedImages.length);
+
+    uploadedImages.forEach((uploadedImage) => {
+      const isNoDuplicates = validateDuplicates(uploadedImage);
```

Comment:
Kompletnie nie podoba mi się że tworzymy zmienne tylko po to aby je wykorzystać raz

Reply:
Przemyślał bym użycie jakieś mapy i iterację po tych funkcjach